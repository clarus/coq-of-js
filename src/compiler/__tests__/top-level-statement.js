// @flow
import {compileAndPrint} from "../index.js";

describe("exports", () => {
  it("handles exports", () => {
    expect(compileAndPrint(`export const x = 12;`)).toMatchSnapshot();
  });
});

describe("imports", () => {
  it("handles React import", () => {
    expect(
      compileAndPrint(`import React, {PureComponent} from "react";`),
    ).toMatchSnapshot();
  });

  it("does not handle non-React imports", () => {
    expect(compileAndPrint(`import express from "express";`)).toMatchSnapshot();
  });
});

describe("definitions", () => {
  it("handles definitions of constants", () => {
    expect(compileAndPrint(`const n = 12;`)).toMatchSnapshot();
  });

  it("handles definitions of functions", () => {
    expect(
      compileAndPrint(`function id<A>(x: A): A {return x;}`),
    ).toMatchSnapshot();
  });

  it("does not handle destructuring of constants", () => {
    expect(compileAndPrint(`const [a, b] = [1, 2];`)).toMatchSnapshot();
  });

  it("does not handle empty declarations", () => {
    expect(compileAndPrint(`let x;`)).toMatchSnapshot();
  });
});
