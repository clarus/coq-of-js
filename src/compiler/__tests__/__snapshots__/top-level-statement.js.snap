// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`definitions does not handle destructuring of constants 1`] = `
"> 1 | const [a, b] = [1, 2];
    |      ^^^^^^

Expected simple identifier"
`;

exports[`definitions does not handle empty declarations 1`] = `
"> 1 | let x;
    |    ^

Expected definition"
`;

exports[`definitions handles definitions of constants 1`] = `
"Definition n :=
  12."
`;

exports[`definitions handles definitions of functions 1`] = `
"Definition id {A : Type} (x : A) : A :=
  x."
`;

exports[`exports handles exports 1`] = `
"Definition x :=
  12."
`;

exports[`imports does not handle non-React imports 1`] = `
"> 1 | import express from \\"express\\";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Only handle imports from React for now"
`;

exports[`imports handles React import 1`] = `""`;
